Mathlib/CategoryTheory/Limits/KanExtension.lean

```diff
@@ -197,6 +208,32 @@ theorem reflective [ι.Full] [ι.Faithful] [∀ X, HasLimitsOfShape (StructuredA
 set_option linter.uppercaseLean3 false in
 #align category_theory.Ran.reflective CategoryTheory.Ran.reflective
 
+variable {D}
+
+/-- The right Kan extension of a functor along its restriction. -/
+abbrev self (F : L ⥤ D) [∀ x, HasLimit (Ran.diagram ι (ι ⋙ F) x)] : L ⥤ D :=
+  Ran.loc ι (ι ⋙ F)
+
+end Ran
+
+/-- A class describing the property of being right Kan extended along a functor. -/
+class IsRightKanExtendedAlong (F : L ⥤ D) [∀ x, HasLimit (Ran.diagram ι (ι ⋙ F) x)] : Prop where
+  /-- There exists an isomorphism to the right kan extension -/
+  isoSelf : Nonempty (F ≅ Ran.self ι F)
```

Comment:
Does an arbitrary isomorphism work? Shouldn't it be `IsIso ((Ran.adjunction D ι).unit.app F)`?

Reply:
I guess it depends on what you mean by "work". It's certainly more natural that way, but it requires the "global" Kan extension to exist. But there still is a canonical map that we can ask to be an iso. Let me mark this as WIP for now and try it out